src/Symfony/Component/Translation/DataCollector/TranslationDataCollector.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\DataCollector;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\DataCollector\LateDataCollectorInterface;
+use Symfony\Component\Translation\DataCollectorTranslator;
+
+/**
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class TranslationDataCollector extends DataCollector implements LateDataCollectorInterface
+{
+    /**
+     * @var DataCollectorTranslator
+     */
+    private $translator;
+
+    /**
+     * @param DataCollectorTranslator $translator
+     */
+    public function __construct($translator = null)
+    {
+        if (null !== $translator && $translator instanceof DataCollectorTranslator) {
+            $this->translator = $translator;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lateCollect()
+    {
+        if (null !== $this->translator) {
+            $this->data = $this->computeCount();
+            $this->data['messages'] = $this->sanitizeCollectedMessages($this->translator->getCollectedMessages());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+    }
+
+    public function getMessages()
+    {
+        return isset($this->data['messages']) ? $this->data['messages'] : null;
```

Comment:
returning an array or null makes it much more complex to deal with. why not returning an empty array ?